arcane/src/arcane/impl/ArcaneBasicMeshSubdividerService.cc

```diff
@@ -183,266 +1055,291 @@ subdivideMesh([[maybe_unused]] IPrimaryMesh* mesh)
-          node_in_cell[new_nodes_on_faces_quatuor[i][1]],
-          node_in_cell[new_nodes_on_faces_quatuor[i][2]],
-          node_in_cell[new_nodes_on_faces_quatuor[i][3]]
-        };
         std::sort(tmp.begin(),tmp.end());
-        Int64 nuid = Arcane::MeshUtils::generateHashUniqueId(tmp.constView());
-        node_in_cell[index_27] = nuid;
-        // Coord on face
-        Arcane::Real3 middle_coord(0.0,0.0,0.0);
-        middle_coord = ( nodes_coordinates[cell.node(new_nodes_on_faces_quatuor[i][0])] + nodes_coordinates[cell.node(new_nodes_on_faces_quatuor[i][1])] + nodes_coordinates[cell.node(new_nodes_on_faces_quatuor[i][2])] + nodes_coordinates[cell.node(new_nodes_on_faces_quatuor[i][3])] ) / 4.0;
-        nodes_to_add_coords[node_in_cell[index_27]] = middle_coord;
-        index_27++;
-      }
+        Int64 uid = MeshUtils::generateHashUniqueId(tmp.constView());
+        node_in_cell[index_27+i] = uid;
 
-      ARCANE_ASSERT((index_27 == 26),("wrong number"));
-
-      Integer new_nodes_on_cell_oct[8] = {0, 1, 5, 4, 3, 2, 7, 6};
-      // Noeud sur cell
-      UniqueArray<Int64> tmp = {
-        node_in_cell[new_nodes_on_cell_oct[0]],
-        node_in_cell[new_nodes_on_cell_oct[1]],
-        node_in_cell[new_nodes_on_cell_oct[2]],
-        node_in_cell[new_nodes_on_cell_oct[3]],
-        node_in_cell[new_nodes_on_cell_oct[4]],
-        node_in_cell[new_nodes_on_cell_oct[5]],
-        node_in_cell[new_nodes_on_cell_oct[6]],
-        node_in_cell[new_nodes_on_cell_oct[7]]
-      };
-
-      // Le noeud central à son uid généré a partir des uid de la cellule parent
-      node_in_cell[index_27] = Arcane::MeshUtils::generateHashUniqueId(tmp.constView());
-
-      // Calcul des coordonnées du noeud central
-      Arcane::Real3 middle_coord(0.0,0.0,0.0);
-      for( Integer i = 0 ; i < 8 ; i++ ){
-        middle_coord += nodes_coordinates[cell.node(new_nodes_on_cell_oct[i])];
-      }
-      middle_coord /= 8.0;
-      nodes_to_add_coords[node_in_cell[index_27]] = middle_coord;
-
-      // Génération des Edges
-      // Pas obligatoire
-      for( Integer i = 8 ; i < 27 ; i++){
-        // Nous calculons plusieurs fois les noeuds pour chaque Cellule (TODO améliorer ça)
-        // Si un noeud n'est pas dans la map, on l'ajoute
-        if( new_nodes.find(node_in_cell[i]) == new_nodes.end()  ){
-          nodes_to_add.add(node_in_cell[i]);
-          new_nodes.insert(node_in_cell[i]);
+
+        if( new_nodes.find(node_in_cell[i+index_27]) == new_nodes.end() ) {
+          // Coords
+          Arcane::Real3 middle_coord(0.0,0.0,0.0);
+          for( Integer j = 0 ; j < node_pattern[i].size() ; j++  ) {
+            info() << "loop" << cell.node(static_cast<Integer>(node_pattern[i][j])) << " " << nodes_coords[cell.node(static_cast<Integer>(node_pattern[i][j]))];
+            middle_coord += nodes_coords[cell.node(static_cast<Integer>(node_pattern[i][j]))];
+          }
+          ARCANE_ASSERT((node_pattern[i].size() != 0),("Wrong size"));
```

Comment:
Use `ARCANE_FATAL` instead with a format string to print some information to help debug (for example the value of `i`)